pkg/api/json_test.go

```diff
@@ -0,0 +1,197 @@
+		want    map[string]interface{}
+		wantErr bool
+	}{
+		{
+			name:    "returns a nil map when JSON is nil",
+			json:    nil,
+			wantErr: false,
+			want:    nil,
+		},
+		{
+			name:    "returns an error when value is not a proper json",
+			json:    JSON{},
+			wantErr: true,
+			want:    map[string]interface{}{},
+		},
+		{
+			name:    "returns an empty map when value an empty json object",
+			json:    JSON([]byte("{}")),
+			wantErr: false,
+			want:    map[string]interface{}{},
+		},
+		{
+			name:    "returns an map of objects from json object",
+			json:    JSON([]byte("{\"a\":\"1\"}")),
+			wantErr: false,
+			want: map[string]interface{}{
+				"a": "1",
+			},
+		},
+	}
+
+	RegisterTestingT(t)
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			object, err := tt.json.Object()
+			Expect(tt.wantErr).To(Equal(err != nil))
+			Expect(tt.want).To(Equal(object))
+		})
+	}
+}
+
+func Test_JSON_MarshalJSON(t *testing.T) {
+	tests := []struct {
+		name    string
+		json    JSON
+		want    driver.Value
+		wantErr bool
+	}{
+		{
+			name:    "Return []byte('null') if json is nil",
+			json:    JSON([]byte(nil)),
+			wantErr: false,
+			want:    []byte("null"),
+		},
+		{
+			name:    "Return json if json is not nil",
+			json:    JSON([]byte("{}")),
+			wantErr: false,
+			want:    []byte("{}"),
+		},
+	}
+
+	RegisterTestingT(t)
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			object, err := tt.json.MarshalJSON()
+			Expect(tt.wantErr).To(Equal(err != nil))
+			Expect(tt.want).To(Equal(object))
+		})
+	}
+}
+
+func Test_JSON_UnmarshalJSON(t *testing.T) {
+	m := JSON([]byte("{\"a\":\"1\"}"))
+	tests := []struct {
+		name    string
+		json    *JSON
+		data    []byte
+		wantErr bool
+	}{
+		{
+			name:    "returns an error when value is not a byte array",
+			json:    nil,
+			wantErr: true,
+		},
+		{
+			name:    "returns nil when a json is passed through.",
+			json:    &m,
+			wantErr: false,
+		},
+	}
+
+	RegisterTestingT(t)
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := tt.json.UnmarshalJSON(tt.data)
+			Expect(tt.wantErr).To(Equal(err != nil))
+		})
+	}
+}
+
+func Test_JSON_Value(t *testing.T) {
+	tests := []struct {
+		name    string
+		json    JSON
+		want    driver.Value
+		wantErr bool
+	}{
+
+		{
+			name:    "Return json if not nil",
+			json:    JSON([]byte("{}")),
+			wantErr: false,
+			want:    []byte("{}"),
+		},
+		{
+			name:    "Return nil if json is nil",
+			json:    JSON([]byte("null")),
+			wantErr: false,
+			want:    nil,
+		},
+		{
+			name:    "Return nil if json is nil",
+			json:    JSON([]byte(nil)),
+			wantErr: false,
+			want:    nil,
+		},
+	}
+
+	RegisterTestingT(t)
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			res, err := tt.json.Value()
+			Expect(tt.wantErr).To(Equal(err != nil))
+			if tt.want == nil {
+				Expect(res).To(BeNil())
+			} else {
+				Expect(tt.want).To(Equal(res))
+			}
```

Comment:
I believe having just
```
Expect(res).To(Equal(tt.want))
```

Moreover, please put the actual value into `Expect` and the expected value into the `Equal...` statement

Reply:
Thanks for the feedback @ziccardi 
This was implemented as such for the same issue as in #933  (refusal to compare nil to nil)

Reply:
Also, thanks for clarifying `expect` and `equal` positioning :) noted

Reply:
yw!